src/main/kotlin/com/terraformation/backend/admin/AdminModulesController.kt

```diff
@@ -24,23 +42,154 @@ import org.springframework.web.servlet.mvc.support.RedirectAttributes
 @RequireGlobalRole([GlobalRole.SuperAdmin, GlobalRole.AcceleratorAdmin])
 @Validated
 class AdminModulesController(
+    private val cohortStore: CohortStore,
     private val deliverablesImporter: DeliverablesImporter,
     private val dslContext: DSLContext,
+    private val eventStore: ModuleEventStore,
+    private val participantStore: ParticipantStore,
+    private val projectStore: ProjectStore,
     private val modulesImporter: ModulesImporter,
+    private val moduleStore: ModuleStore,
 ) {
   private val log = perClassLogger()
+  private val dateFormat = "yyyy-MM-dd'T'HH:mm"
 
   @GetMapping("/modules")
   fun modulesHome(model: Model): String {
     val hasModules = dslContext.fetchExists(MODULES)
+    val cohortNames = cohortStore.findAll().associateBy { it.id }.mapValues { it.value.name }
+    val modules = moduleStore.fetchAllModules()
 
+    model.addAttribute("cohortNames", cohortNames)
     model.addAttribute("canManageDeliverables", currentUser().canManageDeliverables())
     model.addAttribute("canManageModules", currentUser().canManageModules())
     model.addAttribute("hasModules", hasModules)
+    model.addAttribute("modules", modules)
 
     return "/admin/modules"
   }
 
+  @GetMapping("/modules/{moduleId}")
+  fun moduleView(
+      model: Model,
+      @PathVariable moduleId: String,
+      redirectAttributes: RedirectAttributes
+  ): String {
+    val module =
+        try {
+          moduleStore.fetchOneById(ModuleId(moduleId))
+        } catch (e: ModuleNotFoundException) {
+          log.warn("Module not found")
+          redirectAttributes.failureMessage = "Import failed: ${e.message}"
+          return redirectToModulesHome()
+        }
+
+    val moduleProjects = module.cohorts.flatMap { it.projects!!.toList() }
+
+    val cohortNames =
+        module.cohorts.associate { it.cohortId to cohortStore.fetchOneById(it.cohortId).name }
+
+    val projectNames = moduleProjects.associateWith { projectStore.fetchOneById(it).name }
+
+    // cohort name - project name
+    val cohortProjectNames =
+        moduleProjects.associateWith {
+          val project = projectStore.fetchOneById(it)
```

Comment:
Would it make sense to fetch the project models once (maybe store them in a map by ID) and use them for both the project names and the cohort project names? Obviously the list of projects is going to be short but may as well not double-fetch if it's easy to avoid.